### PR TITLE
MULE-19205: Migrate validation in ScatterGatherRouter#initialize

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/CoreComponentBuildingDefinitionProvider.java [hunk 1]
```diff
@@ -186,7 +187,6 @@
   private static final String FLOW = "flow";
   private static final String FLOW_REF = "flow-ref";
   private static final String EXCEPTION_LISTENER_ATTRIBUTE = "exceptionListener";
-  private static final String SCATTER_GATHER = "scatter-gather";
   private static final String PARALLEL_FOREACH = "parallel-foreach";
```
Comment (fsgonz) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/CoreComponentBuildingDefinitionProvider.java:190]: The parallel foreach can also take the one from CoreDslConstants

File: runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/ScatterGatherRoutes.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import org.mule.runtime.ast.api.validation.ValidationResultItem;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+@IgnoreOnLazyInit
+public class ScatterGatherRoutes implements Validation {
+
+  private static final String SCATTER_GATHER_ELEMENT = "scatter-gather";
```
Comment (fsgonz) [inline runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/ScatterGatherRoutes.java:30]: use the one in CoreDslConstants if possible.

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/ScatterGatherRoutes.java (hunk 2 above)
Comment (elrodro83) [reply runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/ScatterGatherRoutes.java:30]: CoreDslConstants is in spring-config, which is not a dependency of this module

Comment (fsgonz) [reply runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/ScatterGatherRoutes.java:30]: Ok!

**Review by fsgonz — APPROVED** [review]

**Pull request merged**